frssci/online/R3BOnlineSpectraFrssci.cxx

```diff
@@ -0,0 +1,291 @@
+/******************************************************************************
+ *   Copyright (C) 2019 GSI Helmholtzzentrum für Schwerionenforschung GmbH    *
+ *   Copyright (C) 2019-2023 Members of R3B Collaboration                     *
+ *                                                                            *
+ *             This software is distributed under the terms of the            *
+ *                 GNU General Public Licence (GPL) version 3,                *
+ *                    copied verbatim in the file "LICENSE".                  *
+ *                                                                            *
+ * In applying this license GSI does not waive the privileges and immunities  *
+ * granted to it by virtue of its status as an Intergovernmental Organization *
+ * or submit itself to any jurisdiction.                                      *
+ ******************************************************************************/
+
+#include "FairLogger.h"
+#include "FairRootManager.h"
+#include "FairRunOnline.h"
+#include "FairRuntimeDb.h"
+#include "TCanvas.h"
+#include "TClonesArray.h"
+#include "TFolder.h"
+#include "TH1.h"
+#include "TH2.h"
+#include "THttpServer.h"
+#include "TMath.h"
+
+#include "R3BEventHeader.h"
+#include "R3BFrssciMappedData.h"
+#include "R3BOnlineSpectraFrssci.h"
+
+R3BOnlineSpectraFrssci::R3BOnlineSpectraFrssci()
+    : R3BOnlineSpectraFrssci("FrssciOnlineSpectra", 1)
+{
+}
+
+R3BOnlineSpectraFrssci::R3BOnlineSpectraFrssci(const char* name, Int_t iVerbose)
+    : FairTask(name, iVerbose)
+    , fMapped(NULL)
+    , fNEvents(0)
+    , fNbDets(1)
+    , fNbPmts(4)
+{
+}
+
+R3BOnlineSpectraFrssci::~R3BOnlineSpectraFrssci()
+{
+    LOG(debug) << "R3BOnlineSpectraFrssci::Destructor";
+    if (fMapped)
+        delete fMapped;
+}
+
+InitStatus R3BOnlineSpectraFrssci::Init()
+{
+    LOG(info) << "R3BOnlineSpectraFrssci::Init()";
+
+    char Name1[255];
+    char Name2[255];
+
+    // try to get a handle on the EventHeader. EventHeader may not be
+    // present though and hence may be null. Take care when using.
+
+    FairRootManager* mgr = FairRootManager::Instance();
+    if (NULL == mgr)
+        LOG(fatal) << "R3BOnlineSpectraFrssci::Init FairRootManager not found";
+
+    fEventHeader = dynamic_cast<R3BEventHeader*>(mgr->GetObject("EventHeader."));
+    if (!fEventHeader)
+        fEventHeader = dynamic_cast<R3BEventHeader*>(mgr->GetObject("R3BEventHeader"));
+
+    FairRunOnline* run = FairRunOnline::Instance();
+    run->GetHttpServer()->Register("", this);
+
+    // --- ------------ --- //
+    // --- MAPPED LEVEL --- //
+    // --- ------------ --- //
+
+    // === get access to mapped data ===//
+    fMapped = dynamic_cast<TClonesArray*>(mgr->GetObject("FrssciMapped"));
```

Comment:
Correct me if I am wrong but should this be
`mgr->GetObject("FrssciMappedData")` ?

Reply:
Yes